<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1193</width>
    <height>787</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QHBoxLayout" name="hlMain">
      <item>
       <widget class="QGroupBox" name="vbControls">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>250</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="title">
         <string>Controls</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QHBoxLayout" name="Sliders">
           <item>
            <layout class="QVBoxLayout" name="Slider1">
             <item>
              <widget class="QLabel" name="labelFOV">
               <property name="text">
                <string>Field of View</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spFOV">
               <property name="alignment">
                <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
               </property>
               <property name="minimum">
                <number>45</number>
               </property>
               <property name="maximum">
                <number>180</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsFOV">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="MinimumExpanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>25</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="font">
                <font>
                 <italic>false</italic>
                </font>
               </property>
               <property name="minimum">
                <number>45</number>
               </property>
               <property name="maximum">
                <number>180</number>
               </property>
               <property name="singleStep">
                <number>0</number>
               </property>
               <property name="pageStep">
                <number>0</number>
               </property>
               <property name="value">
                <number>45</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="tickInterval">
                <number>0</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="Slider2">
             <item>
              <widget class="QLabel" name="labelAngle">
               <property name="text">
                <string>Angle</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spAngle">
               <property name="maximum">
                <number>360</number>
               </property>
              </widget>
             </item>
             <item alignment="Qt::AlignHCenter">
              <widget class="QSlider" name="vsAngle">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="maximum">
                <number>360</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QGroupBox" name="gbProjection">
           <property name="title">
            <string>Projection Mode</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QRadioButton" name="rbOrthogonal">
              <property name="text">
               <string>Orthogonal</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QRadioButton" name="rbPerspective">
              <property name="text">
               <string>Perspective</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="gbRotation">
           <property name="title">
            <string>Rotations</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <layout class="QFormLayout" name="Rotations">
              <item row="0" column="0">
               <widget class="QLabel" name="labelA">
                <property name="text">
                 <string>Ring A</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelB">
                <property name="text">
                 <string>Ring B</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="labelC">
                <property name="text">
                 <string>Ring C</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QSlider" name="hsRotationA">
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSlider" name="hsRotationB">
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSlider" name="hsRotationC">
                <property name="maximum">
                 <number>360</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="gbClipping">
           <property name="title">
            <string>Clipping Pane</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <layout class="QFormLayout" name="Clipping">
              <item row="0" column="0">
               <widget class="QLabel" name="labelNear">
                <property name="text">
                 <string>Near</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelFar">
                <property name="text">
                 <string>Far</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="dsbNear"/>
              </item>
              <item row="1" column="1">
               <widget class="QDoubleSpinBox" name="dsbFar"/>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="cBAnimation">
           <property name="text">
            <string>Animation</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="cBCOG">
           <property name="text">
            <string>Camera on Gimbal</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QFormLayout" name="flControlButton">
           <item row="0" column="0" colspan="2">
            <widget class="QPushButton" name="rbReset">
             <property name="text">
              <string>Reset</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="MyGLWidget" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>480</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>vsFOV</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>83</x>
     <y>108</y>
    </hint>
    <hint type="destinationlabel">
     <x>74</x>
     <y>85</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vsAngle</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>spAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>190</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
